src/docfx/build/xref/InternalXrefSpec.cs

```diff
@@ -29,19 +29,36 @@ public string GetXrefPropertyValue(string propertyName)
 
         public string GetName() => GetXrefPropertyValue("name");
 
-        public ExternalXrefSpec ToExternalXrefSpec(Context context, Document file)
+        public ExternalXrefSpec ToExternalXrefSpec(Context context, Document file, bool forOutput = true)
         {
             var spec = new ExternalXrefSpec
             {
                 Uid = Uid,
                 Monikers = Monikers,
                 Href = Href,
             };
+
+            // DHS appends branch infomation from cookie cache to URL, which is wrong for UID resolved URL
+            // output xref map with URL appending "?branch=master" for master branch
+            if (forOutput)
+            {
+                var (_, query, fragment) = UrlUtility.SplitUrl(Href);
+                var path = DeclairingFile.Docset.Repository.Branch == "master"
```

Comment:
`Docset.Repository` may be `null` when the docset is not managed by a git repository.